functions/src/utils.ts

```diff
@@ -103,3 +100,41 @@ export function isValidUrl(urlString: string | undefined): boolean {
   }
   return url.protocol === "http:" || url.protocol === "https:";
 }
+
+/** Verify a Firebase ID token is valid.
+ *
+ * @param token Firebase ID token to verify.
+ * @returns True if the token is valid, throws an error otherwise.
+ */
+export async function verifyIdToken(token: string | undefined) {
+  if (!token) {
+    throw new ShareLinkError(ShareLinkErrorCode.INVALID_TOKEN);
+  }
+  return admin
+    .auth()
+    .verifyIdToken(token)
+    .then(() => true)
+    .catch((error) => {
+      console.error(error);
+      throw new ShareLinkError(ShareLinkErrorCode.INVALID_TOKEN);
+    });
+}
+
+/** Coerce a value to a boolean or throw an error if not possible.
+ *
+ * @param value The value to coerce.
+ * @returns The coerced boolean value.
+ */
+export function parseBoolean(value: unknown): boolean {
+  if (typeof value === "boolean") return value;
+  if (typeof value === "number" && [0, 1].includes(value)) {
+    return value === 1;
```

Comment:
Does this actually work?  I kind of assume the POST body always gets interpreted as strings, but I'm not 100% sure.  

In general I think I'd be inclined to simplify and only accept "true" and "false" as values or similar.  I don't think we need the flexibility and it could cause confusion, especially if we miss cases like maybe `1` works but `"1"` doesn't.

Reply:
it seemed to work in the tests 😭 but not entirely sure. Either way, I've removed this function and am only allowing `true`/`false` values now